src/object/cli_shard.c

```diff
@@ -218,11 +218,11 @@ dc_rw_cb_csum_verify(const struct rw_cb_args *rw_args)
 	iods_csums = orwo->orw_iod_csums.ca_arrays;
 	maps = orwo->orw_maps.ca_arrays;
 
-	if (daos_obj_is_echo(orw->orw_oid.id_pub))
-		return 0; /** currently don't verify echo classes */
-
-	if (sgls == NULL)
-		return 0; /** no data to verify ... size only */
+	/** currently don't verify echo classes */
+	if ((daos_obj_is_echo(orw->orw_oid.id_pub)) || (sgls == NULL)) {
```

Comment:
Just a thought, why not move csummer_copy below this?

Reply:
This seems to be where the assertion failure is coming from.